### PR TITLE
added moengage as destiantion with identify calls

### PR DESCRIPTION
**Pull request by ruchiramoitra**

### CODE REVIEW

File: __tests__/data/moengage_input.json [hunk 1]
```diff
@@ -0,0 +1,967 @@
+[
```
Comment (arnab-p) [inline __tests__/data/moengage_input.json:1]: Better to use 2-space indentation to keep it similar to others

File: __tests__/data/moengage_output.json [hunk 2]
```diff
@@ -0,0 +1,343 @@
+[
```
Comment (arnab-p) [inline __tests__/data/moengage_output.json:1]: Better to use 2-space indentation to keep it similar to others

File: v0/destinations/moengage/data/MOENGAGEDeviceAttributesConfig.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        "required": true
+    },
+    {
+        "destKey": "model",
+        "sourceKeys": "context.device.model",
+        "required": false
+    },
+    {
+        "destKey": "push_preference",
+        "sourceKeys": "properties.push_preference",
```
Comment (arnab-p) [inline v0/destinations/moengage/data/MOENGAGEDeviceAttributesConfig.json:20]: Lets support `camelCase` as well

File: v0/destinations/moengage/data/MOENGAGEIdentifyAttributesConfig.json [hunk 4]
```diff
@@ -0,0 +1,118 @@
+[
+    {
+        "destKey": "name",
+        "sourceKeys": ["context.traits.name","traits.name"],
```
Comment (arnab-p) [inline v0/destinations/moengage/data/MOENGAGEIdentifyAttributesConfig.json:4]: Traits at top level should get higher preference.

File: v0/destinations/moengage/data/MOENGAGEIdentifyAttributesConfig.json [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+          }
+    },
+    {
+        "destKey": "created_time",
+        "sourceKeys": "originalTimestamp",
+        "required": false
+    },
+    {
+        "destKey": "last_seen",
+        "sourceKeys": ["context.traits.last_seen","traits.last_seen"],
```
Comment (arnab-p) [inline v0/destinations/moengage/data/MOENGAGEIdentifyAttributesConfig.json:86]: Lets support `camelCase` as well

File: v0/destinations/moengage/transform.js [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+  }
+
+  const messageType = message.type.toLowerCase();
+  let category;
+  let response;
+  switch (messageType) {
+    case EventType.IDENTIFY:
+      category = CONFIG_CATEGORIES.IDENTIFY;
+      // only if device information is present device info will be added/updated with an identify call otherwise only user info will be added/updated
+      if (
```
Comment (arnab-p) [inline v0/destinations/moengage/transform.js:97]: Lets build the response initially and then add the device related things later on? Looks like redundant code.

File: v0/destinations/moengage/transform.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+        payload.attributes = flattenJson(payload.attributes); // nested attributes are not shown on dashboard of moengage so it is falttened
+        break;
+      case "device":
+        payload.type = "device";
+        // https://docs.moengage.com/docs/data-import-apis#device-api
+        payload.attributes = constructPayload(
+          message,
+          MAPPING_CONFIG[CONFIG_CATEGORIES.DEVICE_ATTR.name]
+        );
+        payload.attributes = flattenJson(payload.attributes); // nested attributes are not shown on dashboard of moengage so it is falttened
```
Comment (arnab-p) [inline v0/destinations/moengage/transform.js:61]: Move the comments one line above. Line length ideally should be at 80 characters

**Review by arnab-p — CHANGES_REQUESTED** [review]

File: v0/destinations/moengage/transform.js (hunk 7 above)
Comment (ruchiramoitra) [reply v0/destinations/moengage/transform.js:61]: done

File: v0/destinations/moengage/transform.js (hunk 6 above)
Comment (ruchiramoitra) [reply v0/destinations/moengage/transform.js:97]: can you explain a bit more?

File: v0/destinations/moengage/data/MOENGAGEIdentifyAttributesConfig.json (hunk 5 above)
Comment (ruchiramoitra) [reply v0/destinations/moengage/data/MOENGAGEIdentifyAttributesConfig.json:86]: done

File: v0/destinations/moengage/data/MOENGAGEIdentifyAttributesConfig.json (hunk 4 above)
Comment (ruchiramoitra) [reply v0/destinations/moengage/data/MOENGAGEIdentifyAttributesConfig.json:4]: done

File: v0/destinations/moengage/data/MOENGAGEDeviceAttributesConfig.json (hunk 3 above)
Comment (ruchiramoitra) [reply v0/destinations/moengage/data/MOENGAGEDeviceAttributesConfig.json:20]: done

File: __tests__/data/moengage_output.json (hunk 2 above)
Comment (ruchiramoitra) [reply __tests__/data/moengage_output.json:1]: done

File: __tests__/data/moengage_input.json (hunk 1 above)
Comment (ruchiramoitra) [reply __tests__/data/moengage_input.json:1]: done

File: v0/destinations/moengage/transform.js (hunk 6 above)
Comment (ruchiramoitra) [reply v0/destinations/moengage/transform.js:93]: done

**Review by arnab-p — APPROVED** [review]

**Pull request merged**